### ISSUE TITLE
Question about value set management

### ISSUE THREAD

**Issue by qligier** [project-husky/husky#24]
Hey,

I've a question about value set management. We specifically require new type, format and class codes for the eMedication project. Currently, the Java enums are created from ArtDecor value sets, which are copied from the current ordonnances. New codes are needed for implementation, but are quite far from being included in the legal framework. How should we proceed then? I see three possibilities:

1. Manually add them to the enums. Not really clean, since they now contain codes that aren't officially in use.
2. Duplicate these enums in the eMedication package. It's cleaner, but it doesn't work with code that requires the original enums.
3. Create another branch/version that contains the eMedication code and this modification. It requires a bit more work, since it should be kept up-to-date with the main branch.

What do you think?

**Comment by msmock** [comment]
This is a topic we shall discuss in the upcoming meeting of the husky group. I prefer to add a branch with the proprietary updates.

**Comment by project-husky-owner** [comment]
Add version as package to place the new enums in code.

With codegenerator this should be handled correctly.

**Comment by kaisercarole** [comment]
[](url)
[SwissEprValueSetPackage_20221115_beta.txt](https://github.com/project-husky/husky/files/9671946/SwissEprValueSetPackage_20221115_beta.txt)
[SwissEprValueSetPackage_20230415_beta.txt](https://github.com/project-husky/husky/files/9671947/SwissEprValueSetPackage_20230415_beta.txt)
Beta files

**Comment by qligier** [comment]
I've modified the YAML files in the code generator and rerun the generation for the Swiss EPR value sets. The following versions are created:
1. `r202104`: the release 2021-04.
2. `stable`: the last stable release (2021-04 for now)
3. `beta`: the last beta release (2023-04 for now)

The beta release 2022-11 is not generated in the code because all modifications are also contained in `beta`, but we could create a `b202211` version if needed.

I've modified the Husky code that used the old enums and made it use the `stable` version, to avoid changing all the code each time a new release is made. Changes can be seen in #47.

Is it okay for everyone?
